pygeoapi/schemas/config/pygeoapi-config-0.x.yml

```diff
@@ -0,0 +1,472 @@
+                                                  description: format mimetype
+                                          required:
+                                              - name
+                                              - mimetype
+                                      options:
+                                          type: object
+                                          description: optional options key value pairs to pass to provider (i.e. GDAL creation)
+                                          patternProperties:
+                                              "^[a-z]{2}$":
+                                                  allOf:
+                                                      - type: string
+                                      properties:
+                                          type: array
+                                          description: only return the following properties, in order
+                                          items:
+                                              type: string
+                                          minItems: 1
+                                          uniqueItems: true
+                                  required:
+                                      - type
+                                      - name
+                                      - data
+                      required:
+                          - type
+                          - title
+                          - description
+                          - keywords
+                          - extents
+                          - providers
+                    - type: object
+                      description: process object
+                      properties:
+                          type:
+                              type: string
+                              description: resource type
+                              enum:
+                                  - process
+                          processor:
+                              type: object
+                              description: process binding
+                              properties:
+                                  name:
+                                      type: string
+                                      description: |-
+                                          see `pygeoapi.plugin` for supported provider names.
+                                          For custom built plugins, use the import path (e.g. `mypackage.provider.MyProvider`)
+                              required:
+                                  - name
+                      required:
+                          - type
+                          - processor
+
+definitions:
+    multilingual_string:
+        oneOf:
+            - type: string
+            - type: object
+              patternProperties:
+                  "^[a-z]{2}$":
+                      allOf:
+                          - type: string
```

Comment:
Good initiative, @tomkralidis!

However, I would call the type `i18n_string` or something. Technically, the type is language-aware, but does not need to be multi-lingual if the user only sets a single text string and not an object. Unless we decide that this type should always be an object with at least 1 language tag: string key-value pair in it.

Furthermore, the suggested pattern is fine for basic ISO 639-1 language codes (with only 2 chars), but it should also be possible to define localized language tags (e.g. `fr-CA` for Canadian French - see [RFC5646](https://www.rfc-editor.org/rfc/rfc5646.txt)), because the `l10n` module can handle this too. Basically, any code that can be parsed by `Babel` should be valid here, so we should not restrict the pattern too much. The same applies to the array type below, of course.

Reply:
Thanks @GeoSander.  I've renamed the types accordingly.  For patterns, should we allow `^.*$` then or is that too wide?

Reply:
Here is a [complete list of all supported tags](http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry). I think that we can support the majority of the  tags by using `^[a-zA-Z]{2,3}([-_][a-zA-Z0-9]{2,3})?$` perhaps? In other words: 2-3 chars, optionally followed by a dash/underscore and 2-3 digits/chars.

Reply:
Thanks @GeoSander.  Updated with new regexes.